vpr/src/route/rr_graph_indexed_data.cpp

```diff
@@ -264,101 +250,67 @@ static std::vector<size_t> count_rr_segment_types() {
     return rr_segment_type_counts;
 }
 
-static float get_delay_normalization_fac(int nodes_per_chan,
-                                         const t_rr_node_indices& L_rr_node_indices) {
+static float get_delay_normalization_fac() {
     /* Returns the average delay to go 1 CLB distance along a wire.  */
 
-    const int clb_dist = 3; /* Number of CLBs I think the average conn. goes. */
-
-    int inode, itrack, cost_index;
-    float Tdel, Tdel_sum, frac_num_seg;
-
     auto& device_ctx = g_vpr_ctx.device();
+    auto& rr_indexed_data = device_ctx.rr_indexed_data;
 
-    Tdel_sum = 0.;
+    std::vector<float> Tdel_vector(0.0);
+    for (size_t cost_index = CHANX_COST_INDEX_START; cost_index < rr_indexed_data.size(); cost_index++) {
+        float T_value = rr_indexed_data[cost_index].T_linear + rr_indexed_data[cost_index].T_quadratic;
 
-    for (itrack = 0; itrack < nodes_per_chan; itrack++) {
-        inode = find_average_rr_node_index(device_ctx.grid.width(), device_ctx.grid.height(), CHANX, itrack,
-                                           L_rr_node_indices);
-        if (inode == -1)
-            continue;
-        cost_index = device_ctx.rr_nodes[inode].cost_index();
-        frac_num_seg = clb_dist * device_ctx.rr_indexed_data[cost_index].inv_length;
```

Comment:
The `inv_length` doesn't seem to be accounted for anymore?

Reply:
I was unsure about this. Basically, the inv_length is taken into account in a later step of the base_cost calculation and IMO it should be taken into account based on the base_cost_type parameter value.